server/src/main/java/io/crate/planner/operators/MultiPhase.java

```diff
@@ -74,9 +81,41 @@ public ExecutionPlan build(DependencyCarrier executor,
             executor, plannerContext, planHints, projectionBuilder, limit, offset, order, pageSizeHint, params, subQueryResults);
     }
 
+    @Override
+    public List<Symbol> outputs() {
+        return source.outputs();
+    }
+
+    @Override
+    public List<LogicalPlan> sources() {
```

Comment:
This doesn't seem right to me. If this is required to optimise the subqueries, couldn't we add an explicit optimizer pass over the subqueries in the LogicalPlanner or the MultiphaseExecutor? 

Also relates a bit to  https://github.com/crate/crate/pull/15728.
Currently the subqueries get optimized when they're first planned. We don't want to optimise them multiple times.